csharp/extractor/Semmle.Extraction.CSharp.DependencyFetching/DependencyManager.Nuget.cs

```diff
@@ -0,0 +1,364 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Semmle.Util;
+
+namespace Semmle.Extraction.CSharp.DependencyFetching
+{
+    public sealed partial class DependencyManager
```

Comment:
I really like that the logic is moved out of the dependency manager file; Would/could it make sense to have the logic in this file in its own class instead still being a part of the `DependencyManager` class (not as a part of this PR)?

Reply:
Yes, we should do that, this is just a first step.